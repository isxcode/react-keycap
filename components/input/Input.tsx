import React, { useContext } from 'react'
import classNames from 'classnames'
import { ConfigContext } from '../provider/context'
import './style/Input.scss'
import components from '../provider/components'

export type InputCap = 'default' | 'line';

export interface BaseInputProps {
  cap?: InputCap
  label?: string
}

export type InputProps = BaseInputProps & React.InputHTMLAttributes<HTMLInputElement>

const InternalInput: React.ForwardRefRenderFunction<unknown, InputProps> = (props, ref) => {
  const {
    className,
    children,
    label,
    cap
  } = props

  const inputRef = (ref as any) || React.createRef<HTMLElement>()

  const { getPrefixCls } = useContext(ConfigContext)
  const prefixCls = getPrefixCls(components.INPUT)
  const inputCls = classNames(
    prefixCls,
    className,
    {
      [`${prefixCls}-${cap}`]: cap
    }
  )

  const inputNode = (
    <input {...props} className={inputCls} ref={inputRef} placeholder={label}>
      {children}
    </input>
  )

  return <>
    {inputNode}
  </>
}

const Input = React.forwardRef<unknown, InputProps>(InternalInput)

Input.displayName = 'Input'

Input.defaultProps = {
  cap: 'default'
}

export default Input
