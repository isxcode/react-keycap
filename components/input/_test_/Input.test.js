import React from 'react'
import Input from '../Input'
import ReactDOM from 'react-dom'

test('test input component', () => {
  const mountNode = document.createElement('div')

  const inputNode = (
    <Input cap={'default'}/>
  )

  ReactDOM.render(inputNode, mountNode)
})
