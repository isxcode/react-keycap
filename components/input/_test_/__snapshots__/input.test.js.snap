// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button component 1`] = `
<input
  className="keycap-input keycap-input-default"
  which="default"
>
  test
</input>
`;
